subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/userinput/DefaultUserInputHandlerTest.groovy

```diff
@@ -132,23 +150,29 @@ class DefaultUserInputHandlerTest extends Specification {
         input == true
     }
 
-    def "re-requests user input if invalid response to yes/no question"() {
+    def "prompts user again on invalid response to yes/no question"() {
         when:
-        def input = userInputHandler.askYesNoQuestion(TEXT, true)
+        def input = userInputHandler.askBooleanQuestion(TEXT, true)
 
         then:
         1 * outputEventBroadcaster.onOutput(_ as UserInputRequestEvent)
         1 * outputEventBroadcaster.onOutput(_) >> { PromptOutputEvent event -> assert event.prompt == TextUtil.platformLineSeparator }
         1 * outputEventBroadcaster.onOutput(_) >> { PromptOutputEvent event -> assert event.prompt.trim() == 'Accept license? (default: yes) [yes, no]' }
-        1 * userInputReader.readInput() >> 'bla'
-        1 * outputEventBroadcaster.onOutput(_) >> { PromptOutputEvent event -> assert event.prompt.trim() == "Please enter 'yes' or 'no':" }
+        1 * userInputReader.readInput() >> invalidInput
+        1 * outputEventBroadcaster.onOutput(_) >> { PromptOutputEvent event -> assert event.prompt.trim() == "Please enter 'yes' or 'no' (default: 'yes'):" }
         1 * userInputReader.readInput() >> 'no'
         1 * outputEventBroadcaster.onOutput(_) >> { PromptOutputEvent event -> assert event.prompt == TextUtil.platformLineSeparator }
         1 * outputEventBroadcaster.onOutput(_ as UserInputResumeEvent)
         0 * outputEventBroadcaster._
 
         and:
         input == false
+
+        where:
+        invalidInput | _
```

Comment:
💅  The second column is unused, but contains values. Could use `invalidInput << [...]` instead